certification/internal/shell/podman.go

```diff
@@ -222,3 +222,64 @@ func (pe PodmanCLIEngine) Remove(containerID string) (*cli.PodmanRemoveReport, e
 		Stderr: stderr.String(),
 	}, nil
 }
+
+func (pe PodmanCLIEngine) Mount(containerId string) (*cli.PodmanMountReport, error) {
+	var stdout, stderr bytes.Buffer
+	cmd := exec.Command("podman", "mount", containerId)
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	err := cmd.Run()
+	if err != nil {
+		log.Error("could not run mount")
+		return &cli.PodmanMountReport{Stdout: stdout.String(), Stderr: stderr.String()}, err
+	}
+	mountedDir := strings.TrimSpace(stdout.String())
+	return &cli.PodmanMountReport{MountDir: mountedDir, Stdout: stdout.String(), Stderr: stderr.String()}, nil
+}
+
+func (pe PodmanCLIEngine) Unmount(containerId string) (*cli.PodmanUnmountReport, error) {
+	var stdout, stderr bytes.Buffer
+	cmd := exec.Command("podman", "unmount", containerId)
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	err := cmd.Run()
+	report := &cli.PodmanUnmountReport{Stdout: stdout.String(), Stderr: stderr.String()}
+	if err != nil {
+		log.Errorf("could not run unmount. Output: %s", stderr.String())
+		return report, err
+	}
+	return report, nil
+}
+
+func (pe PodmanCLIEngine) Unshare(check, image string, command ...string) (*cli.PodmanUnshareReport, error) {
```

Comment:
We're directly coupling the check names and the image to thsi function signature of the podman engine. If we need to reuse this for any other reason, we would be bound to always having PREFLIGHT_EXEC_CHECK/IMAGE as environment variables., Should we pass an environment instead?

Reply:
I started there, but didn't like that the caller needed to know that it was being used as environment. I preferred to pass them as discreet variables, and leave the environment part as an implementation detail within Unshare.